File: Python/Fundamentals/04_Functions/Exercise/perfect_number.py
number = int(input())


def perfect_num(n):
    sum = 0
    for i in range(1, n):
        if n % i == 0:
            sum += i

    if sum == n and n > 0:
        print('We have a perfect number!')
    else:
        print('It\'s not so perfect.')


perfect_num(number)

File: JS/JS  Fundamentals/04_Functions/exercise/perfectNumber .js
function solve(n) {
    let result = 0

    for (let i = 1; i < n ; i++) {
        if (n % i == 0) {
            result += i;
        }
    }

    if (result == n) {
        return 'We have a perfect number!'
    } else {
        return 'It\'s not so perfect.'
    }
  
}

console.log(solve(6))
console.log(solve(28))
console.log(solve(1236498))